### ISSUE TITLE
PMMLPipeline results different than standard estimator ones

### ISSUE THREAD

**Issue by yairdata** [jpmml/sklearn2pmml#233]
i compared results of the standard lightGBM (without running LabelEncoder before , but indicating the categoricalFeatures so the model was able to conclude them by itself):

```python
model = lgb.LGBMClassifier(max_bin=512,
                                learning_rate=0.001,
                                boosting_type="gbdt",
                                objective="binary",
                                metric="binary_logloss",
                                num_leaves=3,
                                verbose=-1,
                                min_data=50,
                                boost_from_average=True,
                                early_stopping_rounds=50,     
                                verbose_eval=1000,
                               n_estimators=10000)
model.fit(X_train[categorical_columns+numeric_columns+bool_columns],
          y_train,
          early_stopping_rounds=50,
          feature_name=categorical_columns+numeric_columns+bool_columns,
          categorical_feature=categorical_columns,
          eval_set=(X_val[categorical_columns+numeric_columns+bool_columns], y_val))
```
gave:
```
[1138]	valid_0's binary_logloss: 0.00436343
[1139]	valid_0's binary_logloss: 0.00436339
[1140]	valid_0's binary_logloss: 0.00436334
Early stopping, best iteration is:
[1090]	valid_0's binary_logloss: 0.00436181
```

when running it in a pipeline with decorator mapper:
```
mapper = DataFrameMapper(
    [([cat_column], [ CategoricalDomain(),LabelEncoder()]) for cat_column in categorical_columns] +
    [([bool_column], None) for bool_column in bool_columns]+
    [([num_column], ContinuousDomain()) for num_column in numeric_columns]
)
cat_indices = [X_train[categorical_columns+bool_columns+numeric_columns].columns.get_loc(c) for c in categorical_columns if c in X_train]
pipeline_obj = sklearn2pmml.PMMLPipeline([
    ('decorator', mapper),
    ('lgbmc',lgb.LGBMClassifier(
        max_bin=512,
        learning_rate=0.001,
        boosting_type="gbdt",
        objective="binary",
        metric="binary_logloss",
        num_leaves=3,
        verbose=-1,
        min_data=50,
        boost_from_average=True,
        verbose_eval=10,
        n_estimators=10000,
        eval_set=(X_val[categorical_columns+bool_columns+numeric_columns], y_val)
          ))
])
pipeline_obj.fit(X_train[categorical_columns+bool_columns+numeric_columns],
                 y_train,
                 lgbmc__feature_name=categorical_columns+numeric_columns+bool_columns,
                 lgbmc__categorical_feature=cat_indices,
                 lgbmc__early_stopping_rounds=50,
                 lgbmc__eval_set=(X_val[categorical_columns+bool_columns+numeric_columns], y_val)
                )
```
it had worse results:

```
[47]	valid_0's binary_logloss: 0.00462419
[48]	valid_0's binary_logloss: 0.00462426
[49]	valid_0's binary_logloss: 0.00462433
[50]	valid_0's binary_logloss: 0.00462448
[51]	valid_0's binary_logloss: 0.00462454
Early stopping, best iteration is:
[1]	valid_0's binary_logloss: 0.00461958
```

what can be the reason ?

**Comment by vruusmann** [comment]
TLDR: The question is that why `LGBMClassifier.fit(X, y)` and `Pipeline.fit(X, y)` give different results?

You should research this topic on your own, it does not relate to SkLearn2PMML package or JPMML software stack in any way.

Feel free to replace `PMMLPipeline` with standard `sklearn.pipeline.Pipeline`, and you'd still get worse results on the pipeline side.

**Comment by yairdata** [comment]
why does the addition of the pipeline hurt the performance , in any pipeline implementation ?
the pipeline was used just to add the LabelEncoder that was done manually before.
sounds like some argument is not passed in the right manner to the LGBM algorithm.
